src/test/scala/chisel3/tests/AsyncResetSingleBitRegTest.scala

```diff
@@ -0,0 +1,190 @@
+// See LICENSE for license details.
+
+package chisel3.tests
+
+
+import chisel3._
+import chisel3.core.{ExtModule, IntParam}
+import chisel3.tester._
+import firrtl.ir.{Param, Type}
+import org.scalatest._
+import treadle.executable.{PositiveEdge, Transition}
+import treadle.{ScalaBlackBox, ScalaBlackBoxFactory}
+
+class SingleBitRegIO extends Bundle {
+  val in: UInt = Input(UInt(1.W))
+  val out: UInt = Output(UInt(1.W))
+  val enable: Bool = Input(Bool())
+}
+
+/**
+  * This is a black box example that only works with treadle as it does not
+  * define the necessary verilog for verilator/VCS
+  * @param resetValue reset value for this 1 bit register
+  */
+class AsyncResetReg(resetValue: Int = 0) extends ExtModule(Map("RESET_VALUE" -> IntParam(resetValue))) {
+  val io: SingleBitRegIO = IO(new SingleBitRegIO)
```

Comment:
Unless you're using SingleBitRegIO elsewhere (which I don't see), perhaps just inline the bundle def?

Reply:
I agree it's equivalent but what's to be gained by removing it now. And if structural types go away, all the code may go this way anyway